code/modules/hydroponics/hydroponics.dm

```diff
@@ -249,11 +249,13 @@
 		myseed.forceMove(src)
 
 	update_appearance()
-	if((world.time > (lastcycle + cycledelay) && waterlevel > 10 && reagents.total_volume > 2 && pestlevel < 10 && weedlevel < 10) || bio_boosted)
+	if((world.time > (lastcycle + cycledelay) && waterlevel > 10 && (reagents.total_volume > 2 || self_sustaining) && pestlevel < 10 && weedlevel < 10) || bio_boosted)
 		lastcycle = world.time
 		if(myseed && plant_status != HYDROTRAY_PLANT_DEAD)
 			// Advance age
-			age++
+			var/growth_mult = (1.01 ** -myseed.maturation)
+			if(!(age > max(myseed.maturation, myseed.production)/* Fuck Corpse Flowers */ && (growth >= myseed.harvest_age * growth_mult) && self_sustaining))
```

Comment:
Funny, and palpable, but probably not an appropriate comment. Also, what exactly is this accounting for, again? Other than the self-sustaining bit that we talked about.

Reply:
This line only checks if we should bypass aging up on fully grown (self sustaining) plants, with an exception for making sure the age is at least enough for the corpse flower to produce miasma